features/messages/impl/src/main/kotlin/io/element/android/features/messages/impl/timeline/components/reactionsummary/ReactionSummaryView.kt

```diff
@@ -0,0 +1,274 @@
+import androidx.compose.runtime.LaunchedEffect
+import androidx.compose.runtime.derivedStateOf
+import androidx.compose.runtime.getValue
+import androidx.compose.runtime.mutableStateOf
+import androidx.compose.runtime.remember
+import androidx.compose.runtime.rememberCoroutineScope
+import androidx.compose.runtime.saveable.rememberSaveable
+import androidx.compose.runtime.setValue
+import androidx.compose.runtime.snapshotFlow
+import androidx.compose.ui.Alignment
+import androidx.compose.ui.Modifier
+import androidx.compose.ui.graphics.Color
+import androidx.compose.ui.text.style.TextOverflow
+import androidx.compose.ui.tooling.preview.PreviewParameter
+import androidx.compose.ui.unit.dp
+import androidx.compose.ui.unit.sp
+import io.element.android.features.messages.impl.timeline.model.AggregatedReaction
+import io.element.android.libraries.designsystem.components.avatar.Avatar
+import io.element.android.libraries.designsystem.components.avatar.AvatarData
+import io.element.android.libraries.designsystem.components.avatar.AvatarSize
+import io.element.android.libraries.designsystem.preview.DayNightPreviews
+import io.element.android.libraries.designsystem.preview.ElementPreview
+import io.element.android.libraries.designsystem.theme.components.ModalBottomSheet
+import io.element.android.libraries.designsystem.theme.components.Surface
+import io.element.android.libraries.designsystem.theme.components.Text
+import io.element.android.libraries.matrix.api.user.MatrixUser
+import io.element.android.libraries.matrix.ui.model.getAvatarData
+import io.element.android.libraries.theme.ElementTheme
+import kotlinx.coroutines.launch
+
+@OptIn(ExperimentalMaterial3Api::class)
+@Composable
+fun ReactionSummaryView(
+    state: ReactionSummaryState,
+    modifier: Modifier = Modifier,
+) {
+    val sheetState = rememberModalBottomSheetState()
+
+    fun onDismiss() {
+        state.eventSink(ReactionSummaryEvents.Clear)
+    }
+
+    if (state.target != null) {
+        ModalBottomSheet(
+            onDismissRequest = ::onDismiss,
+            sheetState = sheetState,
+            modifier = modifier
+        ) {
+            SheetContent(summary = state.target)
+        }
+    }
+}
+
+@OptIn(ExperimentalFoundationApi::class)
+@Composable
+private fun SheetContent(
+    summary: ReactionSummaryState.Summary,
+    modifier: Modifier = Modifier,
+) {
+    val animationScope = rememberCoroutineScope()
+    var selectedReactionKey: String by rememberSaveable { mutableStateOf(summary.selectedKey) }
+    val selectedReactionIndex: Int by remember {
+        derivedStateOf {
+            summary.reactions.indexOfFirst { it.key == selectedReactionKey }
+        }
+    }
+    val pagerState = rememberPagerState(initialPage = selectedReactionIndex)
+    val reactionListState = rememberLazyListState(initialFirstVisibleItemIndex = selectedReactionIndex)
+
+    LaunchedEffect(pagerState) {
+        snapshotFlow { pagerState.currentPage }.collect { page ->
```

Comment:
Do we need the snapshot flow here? I'd thought using `pagerState.currentPage` directly should be enough.

Reply:
I think this is required to react to changes in  `pager.currentPage`.  it was taken from https://developer.android.com/jetpack/compose/layouts/pager#get-notified.

Reply:
AFAIK with `LaunchedEffect(pagerState.currentPage) { ... }` it should already react to those changes, but I might be mistaken. Is there some account/room where I could test this behaviour?

Reply:
Should be able to test in any room:
1. Add 2 reactions to a mesage
2. Long press on one reaction
3. swipe the list left and right so the pager changes between the lists
4. The highlighted button up top should update.

I think I tested with it and it didn't work.

Reply:
Just re-tested it does in fact work. Will update to that.

Reply:
I did a quick test and I'd say It's working fine. The transition is a bit weird, because moving to the first partially seen item to the end scrolls several items at a time, but moving from the first visible one back to the starts scrolls just 1 position.

Also, just to double check: should the latest reaction group be the first one to appear, on the left side, or the last one?